main.js

```diff
@@ -1 +1,123 @@
+let displayValElement = document.getElementById('calc-display-val');
+
+let displayVal = "0";
+let pendingVal;
+let evalStringArray = [] //hold all operations entered
+
+let calcNumBtns = document.getElementsByClassName('calc-btn-num');
+let calcOperatorBtns = document.getElementsByClassName('calc-btn-operator');
+
+let updateDisplayVal = (clickObj) => {
+    let btnText = clickObj.target.innerText;
+
+    if(displayVal === '0')
+        displayVal = '';
+
+    displayVal += btnText;
+    displayValElement.innerText = displayVal;   
+}
+
+let performOperation = (clickObj) => {
+    let operator = clickObj.target.innerText;
+
+    switch (operator) {
+        case '+':
+            pendingVal = displayVal;
+            displayVal = '0';
+            displayValElement.innerText = displayVal;
+            evalStringArray.push(pendingVal);
+            evalStringArray.push('+');
+            break;
+
+        case '-':
+        pendingVal = displayVal;
+        displayVal = '0';
+        displayValElement.innerText = displayVal;
+        evalStringArray.push(pendingVal);
+        evalStringArray.push('-');
+        break;
+
+        case 'x':
+            pendingVal = displayVal;
+            displayVal = '0';
+            displayValElement.innerText = displayVal;
+            evalStringArray.push(pendingVal);
+            evalStringArray.push('*');
+            break;
+        
+        case '÷':
+        pendingVal = displayVal;
+        displayVal = '0';
+        displayValElement.innerText = displayVal;
+        evalStringArray.push(pendingVal);
+        evalStringArray.push('/');
+        break;
+
+        case '=':
+        evalStringArray.push(displayVal);
+        let evaluation = eval(evalStringArray.join(' ')); 
+        displayVal = evaluation + ''; //result of equation 
+        displayValElement.innerText = displayVal;
+        evalStringArray = [];
+        break;
+    default:
+            break;
+
+
+    }
+}
+
+
+for(let i = 0; i < calcNumBtns.length; i++){
+    calcNumBtns[i].addEventListener('click', updateDisplayVal,false);
+}
+
+//Enables the operations buttons to work
+for (let i = 0; i < calcOperatorBtns.length; i++) {
+    calcOperatorBtns[i].addEventListener('click', performOperation,false);
+}
+
+//Enables the AC/clear button to remove data 
+clearBtn.onclick = () => {
+    displayVal = '0';
+    pendingVal = undefined;
+    evalStringArray = [];
+    displayValElement.innerHTML = displayVal;
+}
+
+//Enables the backpace button to work
+backspaceBtn.onclick = () => {
+    let lengthOfDisplayVal = displayVal.length;
+    displayVal = displayVal.slice(0, lengthOfDisplayVal - 1);
+    
+    if (displayVal === '')
+        displayVal = '0';
+
+    displayValElement.innerText = displayVal;
+}
+
+//Prevents calculator from adding multiple decimals 
+decimalBtn.onclick = () => {
+    if(!displayVal.includes('.'))
+        displayVal += '.';
+    displayValElement.innerText = displayVal;
+}
+
+}
+
+button();
```

Comment:
Great work on this, Kieran! Love the commenting - even more, I loved that you stepped out your comfort zone to work in complex code with the for loops on the button grid. Great job!